tests.py

```diff
@@ -141,24 +141,31 @@ def test_extraction(self):
       self.assertTrue("date" in d2[0])
       self.assertTrue("room_name" in d2[0])
 
+  def testWeather(self):
+    with server.app.test_request_context():
+      res = json.loads(server.weather.retrieve_weather_data().data.decode('utf8'))
+      self.assertTrue(len(res) > 0)
+      s = res['weather_data']
+      self.assertTrue("cod" in s)
```

Comment:
I took it out because the tests failed on Travis without the API key; the attributes do not load when the API key is nonexistent.